### PR TITLE
fix: make pages and techdocs nicely printable / export PDF

### LINKED ISSUE #15754: 🐛 Bug Report: Firefox screenshots do not work beyond "above the fold"
### 📜 Description

Related to the printing issue #15239, Firefox can only screenshot the first visible content on the page and fails on anything below that requires scrolling.

### 👍 Expected behavior

Firefox can screenshot any element, no matter how big it is and if part of the element is not visible on the screen due to scrolling.

### 👎 Actual Behavior with Screenshots

The screenshot will be cutoff to only what is visible rather than the entire, rendered element already on the page.

## Reproduce Backstage demo fail to screenshot

1. Go to https://demo.backstage.io/catalog/default/component/searcher
2. Resize the browser window such that the primary `<article>` within `<main>` is not entirely visible without scrolling
3. Use Firefox's built-in "Take Screenshot" capability to select the `<article>` element and note that the boundaries go below the visible content to cover the entire element
4. The browser may be "frozen" and you may be unable to scroll to the "Download" button because of this issue. So, press "Enter" or "Return" to automatically trigger the "Download" button click
5. View the downloaded image and notice that it is cutoff to only what is visible

## Compare backstage.io screenshot behavior

1. Go to https://backstage.io/
2. Resize the browser window such that any element is partially visible above the scrolling area
3. Use Firefox's built-in "Take Screenshot" capability to select the element and note that the boundaries go below the visible content to cover the entire element
4. Scroll down to the "Download" button. And click the button or press "Enter" or "Return" to download the screenshot as an image.
5. View the downloaded image and notice that the entire element is visible

### 📃 Provide the context for the Bug.

We cannot effectively screenshot larger sections of our Backstage instance without browser plugins that automatically scroll the page and cannot print due to issue #15239.

### 🖥️ Your Environment

- Windows 10 Enterprise and Professional
- Firefox and Chrome (latest evergreen)
- Backstage version:

    Live demo (assuming latest or near latest) as well as an internal version using the dependencies below...

    ```json
    
        "@backstage/app-defaults": "^1.0.10",
        "@backstage/catalog-client": "^1.2.0",
        "@backstage/catalog-model": "^1.1.4",
        "@backstage/cli": "^0.22.0",
        "@backstage/config": "^1.0.5",
        "@backstage/core-app-api": "^1.3.0",
        "@backstage/core-components": "^0.12.2",
        "@backstage/core-plugin-api": "^1.2.0",
        "@backstage/errors": "^1.1.4",
        "@backstage/integration-react": "^1.1.8",
        "@backstage/plugin-api-docs": "^0.8.13",
        "@backstage/plugin-catalog": "^1.7.1",
        "@backstage/plugin-catalog-common": "^1.0.9",
        "@backstage/plugin-catalog-graph": "^0.2.25",
        "@backstage/plugin-catalog-import": "^0.9.3",
        "@backstage/plugin-catalog-react": "^1.2.3",
        "@backstage/plugin-home": "^0.4.29",
        "@backstage/plugin-org": "^0.6.3",
        "@backstage/plugin-permission-react": "^0.4.8",
        "@backstage/plugin-scaffolder": "^1.9.1",
        "@backstage/plugin-search": "^1.0.6",
        "@backstage/plugin-search-common": "^1.2.0",
        "@backstage/plugin-search-react": "^1.3.1",
        "@backstage/plugin-shortcuts": "^0.3.5",
        "@backstage/plugin-stack-overflow": "^0.1.9",
        "@backstage/plugin-tech-radar": "^0.5.20",
        "@backstage/plugin-techdocs": "^1.4.2",
        "@backstage/plugin-user-settings": "^0.6.1",
        "@backstage/theme": "^0.2.16",
        "@backstage/types": "^1.0.2",
    ```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

### PR DESCRIPTION
**Pull request by maapteh**
## Make it printable

fixes #15754 #15239 etc where, because of the page setup, only the first page would be printed. Works for Techdocs and other pages. Bug was caused by the setup of the Layout, which reserved only 100% of the user its view height.

screenshot

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/smart-ads-rush.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/core-components': patch
+---
+
+Make the documentation pages printable (also handy for exporting to pdf when go-green)
```
Comment (freben) [inline .changeset/smart-ads-rush.md:5]: you can use PDF in uppercase to silence this error i think

Comment (freben) [reply .changeset/smart-ads-rush.md:5]: I am not sure about the go-green wording :D `printable (including as PDF)` is clearer to me personally at least!

File: packages/core-components/src/layout/Sidebar/Page.tsx [hunk 2]
```diff
@@ -51,6 +51,10 @@ const useStyles = makeStyles<
       [theme.breakpoints.down('xs')]: {
         paddingBottom: props => props.sidebarConfig.mobileSidebarHeight,
       },
+      '@media print': {
+        paddingLeft: '0px !important',
+        paddingBottom: '0px !important',
```
Comment (freben) [inline packages/core-components/src/layout/Sidebar/Page.tsx:56]: maybe collapse to just `padding` to have it all around?

File: packages/core-components/src/layout/InfoCard/InfoCard.tsx [hunk 3]
```diff
@@ -92,6 +92,10 @@ const VARIANT_STYLES = {
       flexDirection: 'column',
       height: 'calc(100% - 10px)', // for pages without content header
       marginBottom: '10px',
+      breakInside: 'avoid',
```
Comment (freben) [inline packages/core-components/src/layout/InfoCard/InfoCard.tsx:95]: Hm this is NOT inside the media selector. That risks affecting non-print use cases right?

**Review by freben — COMMENTED** [review]
Interesting! I'll have to play around with it to see what the effect is.

File: packages/core-components/src/layout/InfoCard/InfoCard.tsx (hunk 3 above)
Comment (maapteh) [reply packages/core-components/src/layout/InfoCard/InfoCard.tsx]: no, else the rendering will not taken into account, changed into avoid page only a page concept lives in print

Comment (maapteh) [reply packages/core-components/src/layout/InfoCard/InfoCard.tsx]: I could not run the techdocs locally btw, so if you could help me then i can also validate this. I run docker desktop but it still failed...

**Review by freben — APPROVED** [review]
Looks good! @backstage/techdocs-maintainers makes sense to you?

**Review by byan1197 — APPROVED** [review]

**Comment by maapteh** [comment]
^ @soapraj or @agentbellnorm seems you are owner of techdocs package according to OWNERS file?

**Review by freben — APPROVED** [review]
I think let's merge, and then we'll try to get some follow-up later on.

**Comment by maapteh** [comment]
Cool, thanks @freben let me know if i can do follow up work!

**Review by agentbellnorm — COMMENTED** [review]
Sorry for the delay, changes look good to me!

**Comment by kunickiaj** [comment]
Not sure why, but I just get blank pages on 1.15 which included this change, on 1.14 I get some (poorly) rendered text when using the Print dialog in Chrome.

**Comment by maapteh** [comment]
Is it possible to look at this together? I have tested it with the base version

**Pull request merged**